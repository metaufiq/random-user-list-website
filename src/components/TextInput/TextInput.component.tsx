import { Props } from "./TextInput.component.types";

const TextInput = (props: Props)=>{
  return(
    <input 
      className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" 
      id={props.id}
      type={props.type}
      placeholder={props.placeholder}
      onChange={props.onChange}
    />
  )
}

export default TextInput;