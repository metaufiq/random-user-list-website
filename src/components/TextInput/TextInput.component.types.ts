import { ChangeEventHandler, HTMLInputTypeAttribute } from "react"

export type Props = {
  placeholder: string,
  id: string,
  type: HTMLInputTypeAttribute,
  onChange: ChangeEventHandler<HTMLInputElement>
}