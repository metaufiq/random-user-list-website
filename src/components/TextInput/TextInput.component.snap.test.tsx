import { assertSnapshots } from "../../../test";
import TextInput from "./";
import { Props } from "./TextInput.component.types";

describe('TextInputSnapshot', () => {
  const mockProps:Props = {
    onChange:()=>{},
    placeholder: 'placeholder',
    type: 'password'
  };

  const configs = [
    {
      props: mockProps,
      desc: 'should render normal TextInput without id'
    },
    {
      props: { mockProps, id: 'id'},
      desc: 'should render normal TextInput'
    }
  ];

  assertSnapshots(TextInput, configs);
});
