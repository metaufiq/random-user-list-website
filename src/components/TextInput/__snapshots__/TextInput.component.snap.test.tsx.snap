// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputSnapshot should render normal TextInput 1`] = `
<input
  className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
  id="id"
/>
`;

exports[`TextInputSnapshot should render normal TextInput without id 1`] = `
<input
  className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
  onChange={[Function]}
  placeholder="placeholder"
  type="password"
/>
`;
